mlflow/shap.py

```diff
@@ -1,19 +1,95 @@
 from contextlib import contextmanager
 import os
 import tempfile
+import yaml
 
 import numpy as np
 
 import mlflow
+import types
+from mlflow import pyfunc
+from mlflow.exceptions import MlflowException
 from mlflow.utils.annotations import experimental
 from mlflow.utils.uri import append_to_uri_path
+from mlflow.models import Model
 
+from mlflow.tracking.artifact_utils import _download_artifact_from_uri
+from mlflow.models.model import MLMODEL_FILE_NAME
+from mlflow.models.signature import ModelSignature
+from mlflow.models.utils import ModelInputExample, _save_example
+from mlflow.utils.environment import _mlflow_conda_env
+from mlflow.utils.model_utils import _get_flavor_configuration
+from mlflow.protos.databricks_pb2 import RESOURCE_ALREADY_EXISTS
+from mlflow.tracking._model_registry import DEFAULT_AWAIT_MAX_SLEEP_SECONDS
+
+FLAVOR_NAME = "shap"
 
 _MAXIMUM_BACKGROUND_DATA_SIZE = 100
 _DEFAULT_ARTIFACT_PATH = "model_explanations_shap"
 _SUMMARY_BAR_PLOT_FILE_NAME = "summary_bar_plot.png"
 _BASE_VALUES_FILE_NAME = "base_values.npy"
 _SHAP_VALUES_FILE_NAME = "shap_values.npy"
+_UNKNOWN_MODEL_FLAVOR = "unknown"
+_UNDERLYING_MODEL_SUBPATH = "underlying_model"
+
+
+def get_underlying_model_flavor(model):
+    """
+    Find the underlying models flavor.
+
+    :param model: underlying model of the explainer.
+    """
+
+    # checking if underlying model is wrapped
+
+    if hasattr(model, "model"):
+        unwrapped_model = model.model
+
+        # check if passed model is a method of object
+        if isinstance(unwrapped_model, types.MethodType):
+            model_object = unwrapped_model.__self__
+
+            # check if model object is of type sklearn
+            try:
+                import sklearn
+
+                if issubclass(type(model_object), sklearn.base.BaseEstimator):
+                    return "sklearn"
```

Comment:
```suggestion
                    return mlflow.sklearn.FLAVOR_NAME
```

Can we use `mlflow. sklearn.FLAVOR_NAME` here instead of hard-coding `"sklearn"`?

Reply:
and this line too: https://github.com/mlflow/mlflow/pull/3989/files#diff-6ed685f3fee0b5149019a38cbb4cad40a676a5103748d4889818e3c14da4bf0cR66

Reply:
Changed , now using mlflow.sklearn.FLAVOR_NAME and mlflow.pytorch.FLAVOR_NAME